lib/std/zig/system.zig

```diff
@@ -979,20 +927,21 @@ fn detectAbiAndDynamicLinker(
     }
     const ld_info_list = ld_info_list_buffer[0..ld_info_list_len];
 
-    // Best case scenario: the executable is dynamically linked, and we can iterate
-    // over our own shared objects and find a dynamic linker.
-    const elf_file = blk: {
-        // This block looks for a shebang line in /usr/bin/env,
-        // if it finds one, then instead of using /usr/bin/env as the ELF file to examine, it uses the file it references instead,
-        // doing the same logic recursively in case it finds another shebang line.
+    const cwd = fs.cwd();
+    // This block looks for a shebang line in /usr/bin/env,
+    // if it finds one, then instead of using /usr/bin/env as the ELF file to examine, it uses the file it references instead,
+    // doing the same logic recursively in case it finds another shebang line.
 
-        // Since /usr/bin/env is hard-coded into the shebang line of many portable scripts, it's a
-        // reasonably reliable path to start with.
+    // Since /usr/bin/env is hard-coded into the shebang line of many portable scripts, it's a
+    // reasonably reliable path to start with.
+    const elf_file = elf_file: {
+        // Note that shebang line can contain relative path.
         var file_name: []const u8 = "/usr/bin/env";
-        // #! (2) + 255 (max length of shebang line since Linux 5.1) + \n (1)
+        // #! (2) + 255 (max length of shebang line since Linux 5.1) + \n (1) just in case.
+        // See `man execve(2)`.
         var buffer: [258]u8 = undefined;
         while (true) {
-            const file = fs.openFileAbsolute(file_name, .{}) catch |err| switch (err) {
+            const file = cwd.openFile(file_name, .{ .mode = .read_only }) catch |err| switch (err) {
```

Comment:
`openFileAbsolute` is correct when the file path is an absolute path, as it is with `/usr/bin/env` as well as any shebang line encountered.